### PR TITLE
`Development`: Refactor tutorial group tests

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a [very good performance](https://docs.artemis.cit.tum.de/dev/guidelines/performance/) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Description

Introduces `AbstractTutorialGroupIntegrationTest`.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unchanged

## Summary by CodeRabbit

- **New Features**
	- Introduced a new test class for validating the architecture of the tutorial group module.
- **Bug Fixes**
	- Updated test methods to ensure correct behavior regarding user permissions and tutorial group functionalities.
- **Refactor**
	- Restructured existing test classes to enhance organization and accessibility of dependencies.
	- Removed certain dependencies from `TutorialGroupIntegrationTest` to streamline the testing process.

### CODE REVIEW

File: src/test/java/de/tum/cit/aet/artemis/tutorialgroup/AbstractTutorialGroupIntegrationTest.java [hunk 1]
```diff
@@ -49,58 +51,67 @@
 import de.tum.cit.aet.artemis.tutorialgroup.test_repository.TutorialGroupTestRepository;
 import de.tum.cit.aet.artemis.tutorialgroup.util.TutorialGroupUtilService;
 
-/**
- * Contains useful methods for testing the tutorial groups feature.
- */
-abstract class AbstractTutorialGroupIntegrationTest extends AbstractSpringIntegrationLocalCILocalVCTest {
+public abstract class AbstractTutorialGroupIntegrationTest extends AbstractSpringIntegrationIndependentTest {
```
Comment (JohannesStoehr) [inline src/test/java/de/tum/cit/aet/artemis/tutorialgroup/AbstractTutorialGroupIntegrationTest.java:54]: Why is this class public now?

**Review by JohannesStoehr — APPROVED** [review]
Approve refactoring

File: src/test/java/de/tum/cit/aet/artemis/tutorialgroup/AbstractTutorialGroupIntegrationTest.java (hunk 1 above)
Comment (JohannesStoehr) [reply src/test/java/de/tum/cit/aet/artemis/tutorialgroup/AbstractTutorialGroupIntegrationTest.java:54]: Answer: it is needed in the test files that may lay in another package

**Review by BBesrour — APPROVED** [review]
code

**Review by iyannsch — APPROVED** [review]
Code LGTM

**Review by cremertim — APPROVED** [review]

**Pull request merged**